recipe/meta.yaml

```diff
@@ -1,35 +1,46 @@
-{% set version = "3.2.3" %}
+{% set version = "3.2.4.post1" %}
 
 package:
   name: scs
   version: {{ version }}
 
 source:
   - url: https://github.com/bodono/scs-python/archive/refs/tags/{{ version }}.tar.gz
-    sha256: 7592805810fb6da09955dc45322b2d376bca993928c972f472f1a5352821ddcd
+    sha256: bb509507d857906eb95d17d5824138e48e7c75d1a692d0aa7256d899c7e703c5
   # include submodule (not in github tarball due to dear-github/dear-github#214)
   - git_url: https://github.com/cvxgrp/scs.git
-    git_rev: ac6840a3b3264950e6c300264cbf3937e0bcc6c5
-    folder: scs
+    git_rev: 9024b8ccc1bba6ee797440fb22354cadb9c81839
+    folder: scs_source
 
 build:
   number: 0
   script:
-    - export BLAS_LAPACK_LIB_PATHS="$PREFIX/lib"    # [unix]
-    - export BLAS_LAPACK_LIBS="lapack:cblas:blas"   # [unix]
-    - {{ PYTHON }} -m pip install . -vv
+    # Copy Meson args split from pandas feedstock but split on cross-compiling
+    # instead of unix/win because Linux aarch64 does not work with the Meson
+    # args modification (fails to find blas/lapack)
+    #
+    # See https://github.com/conda-forge/pandas-feedstock/blob/ea054ffba9cda5f10436e1c7250e62d7ee1aacd1/recipe/meta.yaml#L20-L22
+    - MESON_ARGS_REDUCED="$(echo $MESON_ARGS | sed 's/--buildtype release //g')"              # [not aarch64 and build_platform != target_platform]
+    - {{ PYTHON }} -m pip install . -vv -Csetup-args=${MESON_ARGS_REDUCED// / -Csetup-args=}  # [not aarch64 and build_platform != target_platform]
+    - {{ PYTHON }} -m pip install . -vv                                                       # [aarch64 or build_platform == target_platform]
```

Comment:
What I meant is that we should explicitly tell meson to look for the right lapack:
```
    -Csetup-args=-Dblas=blas \
    -Csetup-args=-Dlapack=lapack \
```
cf. scipy's [`build.sh`](https://github.com/conda-forge/scipy-feedstock/blob/main/recipe/build.sh). I don't want to special case aarch64 here without trying to fix it properly first.

Reply:
I believe that osx_arm64 is the special case. From what I saw, ppc64le worked with the MESON_ARGS_REDUCED commands as well as the simple `pip install .`. So we could rewrite the selectors as `osx and arm64` and `not (osx and arm64)`. Would that be better?

There are a few things I am unclear on currently. With the current configuration, I see for all builds that I checked:

```
  Library blas found: NO
  Run-time dependency blas found: YES 3.9.0
  Library lapack found: NO
  Run-time dependency lapack found: YES 3.9.0
  Library cblas found: NO
  Run-time dependency cblas found: YES 3.9.0
```

Are the NO answers here an issue? It does find the run-time dependencies and build and pass the tests. I tried adding the blas and lapack options [here](https://github.com/conda-forge/scs-feedstock/commit/60f4907ed33bcbbda44e81e7d304ba7da13f036c) but got

```
../meson.build:1:0: ERROR: Unknown options: "blas, lapack"
```

Maybe I did not do it right? I interpreted that to mean those were meson config options that scs-python had not set up.

I don't fully understand what `MESON_ARGS_REDUCED` is doing. From comment in the pandas feedstock (really more [this comment](https://github.com/conda-forge/scipy-feedstock/blob/9401aa4ee5d9a22c2c9d219e5fb6dbd580f30207/recipe/build.sh#L12-L14) in the scipy feedstock), I took it that the buildtype flag was already inserted into `MESON_ARGS` by meson-python and having it included twice was problematic so it needs to be removed. However, if I don't use the `not aarch64` selector (like [here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=862809&view=logs&jobId=e9afaa34-1a0f-534e-78fc-fae528ccd915&j=e9afaa34-1a0f-534e-78fc-fae528ccd915&t=869ab379-d63c-50ae-655f-69018fb07995)), then I see for aarch64

```
  Library blas found: NO
  Run-time dependency blas found: NO (tried pkgconfig and cmake)
  Library lapack found: NO
  Run-time dependency lapack found: NO (tried pkgconfig and cmake)
  Library cblas found: NO
  Run-time dependency cblas found: NO (tried pkgconfig and cmake)
```

and the aarch64 build fails completely. I don't understand what failing to find blas/lapack has to do with filtering `buildtype` out of the meson args.

Also, if I don't use the `MESON_ARGS_REDUCED` method to filter out `buildtype` (like [here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=863496&view=logs&jobId=1b8be447-c2bd-5772-b66a-a1146441bf88&j=1b8be447-c2bd-5772-b66a-a1146441bf88&t=0da7bde9-21e8-5d19-f418-738876121ca7)), the osx_arm64 build fails with:

  ../meson.build:1:0: ERROR: Could not invoke sanity test executable: [Errno 86] Bad CPU type in executable: '/Users/runner/miniforge3/conda-bld/scs_1706040399741/work/.mesonpy-n38da1p9/meson-private/sanitycheckc.exe'.
```

That seems like the cross-compilation is not configured properly and a build meson executable is getting used instead of the host one, but I don't see what that has to do with filtering out the `buildtype` (unless cross-compilation is an option for `buildtype`, but it seems like it is for release/debug).

Reply:
> That seems like the cross-compilation is not configured properly and a build meson executable is getting used instead of the host one, but I don't see what that has to do with filtering out the `buildtype` (unless cross-compilation is an option for `buildtype`, but it seems like it is for release/debug).

Most likely you need something like [this](https://github.com/conda-forge/scipy-feedstock/blob/main/recipe/build.sh#L6-L10).

> Are the NO answers here an issue? It does find the run-time dependencies and build and pass the tests. I tried adding the blas and lapack options [here](https://github.com/conda-forge/scs-feedstock/commit/60f4907ed33bcbbda44e81e7d304ba7da13f036c) but got
> 
> ```
> ../meson.build:1:0: ERROR: Unknown options: "blas, lapack"
> ```

Not 100% sure if this needs handling in scs's meson files. CC the people involed in https://github.com/bodono/scs-python/pull/70: @enzbus @rgommers @bodono

> However, if I don't use the `not aarch64` selector (like [here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=862809&view=logs&jobId=e9afaa34-1a0f-534e-78fc-fae528ccd915&j=e9afaa34-1a0f-534e-78fc-fae528ccd915&t=869ab379-d63c-50ae-655f-69018fb07995)), then I see for aarch64 [...]

I haven't had time to dig yet, but my point is: the only distinction in the builds here should be between native compilation and cross-compilation, not between linux-aarch64 & osx-arm64.

Reply:
The NO answer by scs-python meson file are not an issue, if it compiles and passes tests it's all good. (We have a few library searches to catch various configurations.)

Reply:
And for osx, it should link to the system libraries (Accelerate) automatically, both amd64 and arm64...

Reply:
PS @wshanks , it looks like we didn't implement in the `meson.option` an arbitrary search path for (open)blas, feel free to open a PR in scs-python if you need that. Here https://github.com/bodono/scs-python/blob/dd17e2e5282ebe85f2df8a7c6b25cfdeb894970d/meson.build#L42 we hard-code a path that's used in github CI to build windows wheels. Maybe we can add another block with a similar syntax for the path you need. I'm not an expert in meson, from a quick search it appears you can export search paths directly to the compiler if you need -> https://stackoverflow.com/questions/61058722/how-can-i-specify-library-path-when-using-meson . Not sure if that helps, let me know!

Reply:
@enzbus I don't think passing a specific path should be necessary. Looking at scipy, I see that it just uses setup options like `-Dblas=blas` here:

https://github.com/conda-forge/scipy-feedstock/blob/9401aa4ee5d9a22c2c9d219e5fb6dbd580f30207/recipe/build.sh#L17-L20

but scs should already be trying `blas=blas` and `lapack=lapack` here:

https://github.com/bodono/scs-python/blob/dd17e2e5282ebe85f2df8a7c6b25cfdeb894970d/meson.build#L10-L25

For reference the `blas` and `lapack` options get defined for scipy:

https://github.com/scipy/scipy/blob/e8b043df1d586d9df9fed96add1a536859f1cb8c/meson_options.txt#L1-L4

and then used around here:

https://github.com/scipy/scipy/blob/e8b043df1d586d9df9fed96add1a536859f1cb8c/scipy/meson.build#L165

The usage is ultimately just calling `dependency('blas')` which is what scs does as well, so I think the look up should be the same as for scipy.

The only significant differences that I see between this feedstock and the scipy feedstock currently are that the scipy feedstock includes pkgconfig (the blas lookup failure mentions that it tried using pkgconfig) and the scipy feedstock uses `python -m build` instead of `pip install`. I think this is because pip used to take only the last `-C` option rather than concatenating them. There is a warning about this [here](https://meson-python.readthedocs.io/en/latest/how-to-guides/config-settings.html) but the linked issue in pip was closed as resolved last year. I already tried adding pkgconfig before and did not see a change. Maybe there was some other issue at the time. I will try adding pkgconfig again and using `python -m build` to see if they help.

I am also confused about why the build works for aarch64 without taking `--buildtype release` out of the MESON_ARGS. For the PyPy ppc issue commented on elsewhere in the PR, I tried printing out `importlib.machinery.EXTENSION_SUFFIXES` and for aarch64 it printed out an x86_64 suffix which made me confused about how the cross-compile is set up. For ppc, the suffix was a ppc suffix, not x86.

Reply:
Installing `pkgconfig` should make `dependency('blas')` work in conda-forge.